node_modules/gh-auth/lib/oauth.js

```diff
@@ -0,0 +1,516 @@
+ * @param  {OauthClient}    callback.client         The updated oauth client
+ */
+var updateClient = module.exports.updateClient = function(ctx, clientId, fields, callback) {
+    clientId = GrasshopperUtil.getNumberParam(clientId);
+    fields = fields || {};
+
+    var validator = new Validator();
+    validator.check(null, {'code': 401, 'msg': 'Only administrators can update a client'}).isAdmin(ctx, ctx.app.id);
+    validator.check(clientId, {'code': 400, 'msg': 'An integer client id must be specified'}).isInt();
+    validator.check(_.keys(fields).length, {'code': 400, 'msg': 'At least one field to update must be specified'}).min(1);
+
+    if (fields.displayName) {
+        validator.check(fields.displayName, {'code': 400, 'msg': 'A display name can be at most 255 characters long'}).isShortString();
+    }
+
+    if (fields.redirectUri) {
+        validator.check(fields.redirectUri, {'code': 400, 'msg': 'A valid redirect URI must be specified'}).isUrl();
+    }
+
+    if (validator.hasErrors()) {
+        return callback(validator.getFirstError());
+    }
+
+    AuthOauthDAO.getClient(clientId, function(err, client) {
+        if (err) {
+            return callback(err);
+        }
+
+        validator.check(null, {'code': 401, 'msg': 'Only administrators can update a client'}).isAdmin(ctx, client.AppId);
+        if (validator.hasErrors()) {
+            return callback(validator.getFirstError());
+        }
+
+        return AuthOauthDAO.updateClient(client, _.ghCompact(fields), callback);
+    });
+};
+
+/**
+ * Disable an oauth client
+ *
+ * @param  {Context}        ctx                 The context of the current request
+ * @param  {Number}         clientId            The id of the client
+ * @param  {Function}       callback            Standard callback function
+ * @param  {Object}         callback.err        An error that occurred, if any
+ * @param  {OauthClient}    callback.client     The disabled oauth client
+ */
+var disableClient = module.exports.disableClient = function(ctx, clientId, callback) {
+    clientId = GrasshopperUtil.getNumberParam(clientId);
+
+    var validator = new Validator();
+    validator.check(null, {'code': 401, 'msg': 'Only administrators can disable a client'}).isAdmin(ctx, ctx.app.id);
+    validator.check(clientId, {'code': 400, 'msg': 'An integer client id must be specified'}).isInt();
+    if (validator.hasErrors()) {
+        return callback(validator.getFirstError());
+    }
+
+    AuthOauthDAO.getClient(clientId, function(err, client) {
+        if (err) {
+            return callback(err);
+        }
+
+        validator.check(null, {'code': 401, 'msg': 'Only administrators can disable a client'}).isAdmin(ctx, client.AppId);
+        if (validator.hasErrors()) {
+            return callback(validator.getFirstError());
+        } else if (client.disabled) {
+            return callback({'code': 400, 'msg': 'The specified client is already disabled'});
```

Comment:
Can we do `return callback();` here? I think it might make things easier in the UI